Sources/SemanticAnalyzer/SemanticError.swift

```diff
@@ -108,6 +108,12 @@ extension Diagnostic {
   static func renderCapabilityGroup(_ capabilities: [CallerCapability]) -> String {
     return "\(capabilities.map({ $0.name }).joined(separator: ", "))"
   }
+
+  static func recursiveStructs(_ identifier: Identifier, _ root: PropertyInformation) -> Diagnostic {
```

Comment:
Ideally this would go after `statePropertyUsedWithinPropertyInitializer`, and would be named `recursiveStruct(_ structIdentifier: _ enclosingType:)`